android/src/main/java/com/plaid/PlaidModule.kt

```diff
@@ -224,7 +201,16 @@ class PlaidModule internal constructor(reactContext: ReactApplicationContext) :
           .emit("onEvent", convertJsonToMap(JSONObject(json)))
       }
 
-      val tokenConfiguration = getLinkTokenConfiguration(data)
+      val obj = JSONObject(data)
+      val publicKey = maybeGetStringField(obj, PUBLIC_KEY)
+      // If we're initializing with a Link token, we will not use or
+      // accept many of the client-side configs.
+      val token = maybeGetStringField(obj, TOKEN)
+      if (publicKey == null && token == null) {
+        throw IllegalStateException("Token must be part of configuration.")
+      }
+
+      val tokenConfiguration = getLinkTokenConfiguration(obj, token)
```

Comment:
Actually, instead of trying to create the LinkPublicKeyConfig first, let's attempt to create the publicKeyConfig and then when public_key doesn't exist or something, we try to create LinkTokenConfig